{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fedb6831-03c1-4345-864a-cc05247beed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import altair as alt\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6c165bd4-bdcf-498c-a0cd-d6de3d441f9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv('./huanan-market-2024/metadata/Liu_etal_2023_with_sequencing.csv')\n",
    "a = pd.read_csv('./huanan-market-2024/mitochondrial_mappings/mitochondrial_metazoa_counts_93.tsv', sep=\"\\t\")\n",
    "a2 = pd.read_csv('./huanan-market-2024/mitochondrial_mappings/mitochondrial_metazoa_coveredbases_93.tsv', sep=\"\\t\")\n",
    "\n",
    "a = a[['Lab code', 'Run', 'Sample_category', 'Stall_corrected'] + list(a2[a2.columns[5:]].loc[:, a2[a2.columns[5:]].sum() >= 1000].columns)]\n",
    "\n",
    "a2 = a2[['Lab code', 'Run', 'Sample_category', 'Stall_corrected'] + list(a2[a2.columns[5:]].loc[:, a2[a2.columns[5:]].sum() >= 1000].columns)]\n",
    "\n",
    "\n",
    "data = a.melt(id_vars=['Lab code', 'Run', 'Sample_category', 'Stall_corrected'],value_vars=a.columns[5:], value_name='read_counts')\n",
    "data2 = a2.melt(id_vars=['Lab code', 'Run', 'Sample_category', 'Stall_corrected'],value_vars=a.columns[5:], value_name='covered_bases')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f31a287-9ab9-4144-aef6-32e3efad6f2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = pd.merge(data,data2, on=['Run','variable'])\n",
    "\n",
    "sample_types = {\"market_stall_wildlife\":\"Wildlife_stalls_Jan12\",\n",
    "\"market_west_Jan1\": \"Non-wildlife_stalls_Jan1\",\n",
    "\"market_825_late\":\"Wildlife_stall_B_825\",\n",
    "\"market_west_drain\":\"Drain\",\n",
    "\"market_sewage_west\":\"NA\",\n",
    "\"market_629_Jan12\":\"Wildlife_stall_A_629\",\n",
    "\"market_west_late\":\"All_stalls_AfterJan12\",\n",
    "\"market_629_late\":\"Wildlife_stall_A_629\",\n",
    "\"non_market_sewage\":\"NA\",\n",
    "\"market_825\":\"Wildlife_stall_B_825\",\n",
    "\"market_sewage_east\":\"NA\",\n",
    "\"market_east_drain\":\"Drain\",\n",
    "\"warehouse\":\"All_stalls_AfterJan12\",\n",
    "\"market_east_late\":\"All_stalls_AfterJan12\",\n",
    "\"market_east_Jan1\":\"Non-wildlife_stalls_Jan1\",\n",
    "\"market_stall_wildlife_late\":\"All_stalls_AfterJan12\"}\n",
    "\n",
    "final['new_category'] = final.Sample_category_x.map(sample_types)\n",
    "old_final = final\n",
    "final = final.query(\"new_category != 'NA'\") ## Removes Non-market samples\n",
    "\n",
    "seq_data = pd.read_csv('./huanan-market-2024/metadata/Sequencing_run_info.tsv', sep=\"\\t\")\n",
    "not_included = seq_data[(seq_data.Sample_Type == 'sars2_amplicon') | (seq_data.Sample_category == 'single_end_duplicate')].Run.to_list()\n",
    "\n",
    "final = final[~final.Run.isin(not_included)]\n",
    "\n",
    "final.new_category = pd.Categorical(final.new_category, [\"Wildlife_stall_A_629\", \"Wildlife_stall_B_825\", \"Wildlife_stalls_Jan12\",\n",
    "                                                        'Non-wildlife_stalls_Jan1', 'All_stalls_AfterJan12', 'Drain'])\n",
    "final = final.sort_values(\"new_category\")\n",
    "\n",
    "sample_id_to_date = dict(zip(d['Lab.code'], d['Sampling.date']))\n",
    "final['Sampling_date'] = final['Lab code_x'].map(sample_id_to_date)\n",
    "\n",
    "# print(\"Removed samples:\")\n",
    "# old_final[~old_final.Run.isin(list(final.Run))].groupby(\"Run\").first()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "25ba7dce-9d98-47c7-b435-8eab7d8777af",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "### Get SARS2 read info\n",
    "\n",
    "sars2 = pd.read_csv('./huanan-market-2024/sars2_mappings/sars2_reads_post_trimming.tsv', sep=\"\\t\")\n",
    "sars2.columns = ['Run','SARS2_read_count','SARS2_covered_bases']\n",
    "\n",
    "\n",
    "# sars2.plot.scatter(\"SARS2_covered_bases\",\"SARS2_read_count\")\n",
    "# plt.yscale(\"log\")\n",
    "\n",
    "# plt.show()\n",
    "\n",
    "sars2_seqs = pd.merge(sars2, d[d.Sequencing_Run.isin(final.Run)][['Sample.ID', 'Lab.code','Sequencing_Run','SARS.CoV.2.qPCR.result']], left_on='Run', right_on='Sequencing_Run')\n",
    "\n",
    "pcr = pd.read_csv('./huanan-market-2024/metadata/Liu_PCR_results.csv')\n",
    "\n",
    "sars2_seqs = pd.merge(sars2_seqs, pcr, left_on='Sample.ID', right_on='Sample ID')\n",
    "\n",
    "sars2_seqs = sars2_seqs[['Run','SARS2_read_count','SARS2_covered_bases','PCR','CT']]\n",
    "\n",
    "final = pd.merge(final, sars2_seqs, on='Run', how='left')\n",
    "\n",
    "final['SARS2_positive'] = 'Negative'\n",
    "final.loc[(final.PCR == '+') | (final.SARS2_covered_bases > 0), 'SARS2_positive'] = 'Positive'\n",
    "\n",
    "final['Contig'] = final['variable'].str.split(expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ef9f88fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "species = pd.read_csv('./huanan-market-2024/mitochondrial_mappings/species_descriptions_with_common_name.csv')\n",
    "species = species[['Contig','Species','Common_name', 'Group']]\n",
    "species['Common_name'] = species['Common_name'].str.capitalize()\n",
    "species['Common_and_scientific'] = species['Common_name'] + \" (\" + species['Species'] + \")\"\n",
    "species['is_mammal'] = False\n",
    "species.loc[species.Group == 'Mammal','is_mammal'] = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2bdbdb16",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = pd.merge(final, species, on='Contig')\n",
    "\n",
    "final.columns = ['Sample lab code', 'Sequencing run', 'Sample category old', 'Stall', 'variable', 'Read counts',\n",
    "                 'Lab code_y', 'Sample_category_y', 'Stall_corrected_y', 'MT Covered bases (bp)', 'Sample category',\n",
    "                 'Sampling date','SARS2 reads count', 'SARS2 covered bases', 'SARS2 PCR', 'CT', 'SARS2 positivity',\n",
    "                 'Contig', 'Species', 'Common name', 'Group', 'Common_and_scientific', 'is_mammal']\n",
    "final = final.query(\"`Read counts` > 0\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "48aa823b-f0f5-4fc3-9369-48ead3fb90ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "alt.data_transformers.enable(\"vegafusion\")\n",
    "\n",
    "facet_order = [\"Wildlife_stall_A_629\", \"Wildlife_stall_B_825\", \"Wildlife_stalls_Jan12\",\n",
    "                                                        'Non-wildlife_stalls_Jan1', 'All_stalls_AfterJan12', 'Drain']\n",
    "\n",
    "\n",
    "source = final\n",
    "source['Log(MT Read count)'] = np.log10(source['Read counts'] + 1)\n",
    "\n",
    "source = source.query(\"is_mammal == True\")\n",
    "\n",
    "\n",
    "#### Animal selection\n",
    "options = list(source.Common_and_scientific.sort_values().unique())\n",
    "\n",
    "input_dropdown = alt.binding_select(\n",
    "    options=options + [None],\n",
    "    labels=options + ['All'],\n",
    "    name='Species: '\n",
    ")\n",
    "selection = alt.selection_point(\n",
    "    fields=['Common_and_scientific'],\n",
    "    bind=input_dropdown,\n",
    ")\n",
    "\n",
    "\n",
    "#### SARS2 selection\n",
    "input_dropdown2 = alt.binding_radio(\n",
    "    options=['Positive',None],\n",
    "    labels=['Positive Only','All'],\n",
    "    name='SARS2 detection status: '\n",
    ")\n",
    "selection2 = alt.selection_point(\n",
    "    fields=['SARS2 positivity'],\n",
    "    bind=input_dropdown2,\n",
    ")\n",
    "\n",
    "#### Sampling dates selection\n",
    "options2 = list(final['Sampling date'].unique())\n",
    "\n",
    "input_dropdown3 = alt.binding_radio(\n",
    "    options=[None] + options2,\n",
    "    labels=['All'] + options2 ,\n",
    "    name='Sampling dates: ',\n",
    ")\n",
    "\n",
    "selection3 = alt.selection_point(\n",
    "    fields=['Sampling date'],\n",
    "    bind=input_dropdown3\n",
    ")\n",
    "\n",
    "alt.Chart(source).mark_circle(size=80).encode(\n",
    "    x='MT Covered bases (bp)',\n",
    "    y='Log(MT Read count)',\n",
    "    color='Common name',\n",
    "    tooltip=['Sample lab code', 'Sample category', 'Stall', 'Sequencing run', 'Common name', 'Species', 'Read counts', 'MT Covered bases (bp)', 'SARS2 reads count','SARS2 covered bases','SARS2 PCR']\n",
    ").interactive().add_params(\n",
    "    selection\n",
    ").add_params(selection2).add_params(selection3).transform_filter(\n",
    "    selection\n",
    ").transform_filter(\n",
    "    selection2).transform_filter(\n",
    "    selection3).facet(\n",
    "    facet=alt.Facet('Sample category', sort=facet_order),\n",
    "    columns=3\n",
    ").save(\"mammals.html\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd1ec91e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base_conda",
   "language": "python",
   "name": "base_conda"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
